test/powershell/engine/Api/BasicEngine.Tests.ps1

```diff
@@ -47,3 +47,46 @@ exit
         }
     }
 }
+
+Describe "EndInvoke() should return a collection of results" -Tags "CI" {
+    BeforeAll {
+        $ps = [powershell]::Create()
+        $ps.AddScript("'Hello'; Sleep 1; 'World'") > $null
```

Comment:
> Sleep 1

We should use `Wait-UntilTrue` and use flags to sync threads.

Reply:
@daxian-dbw I also don't like sleep timeouts in tests.

Reply:
Thank you both for the comments. The `Sleep 1` is not really necessary, so I removed it.